### DISCUSSION TITLE
How to force VS Code auto-import suggestions to use path without `lib` prefix?

### DISCUSSION THREAD

**Discussion by onvlt** [gcanti/fp-ts#1709] [Q&A]
I'd like to use `fp-ts/Module`, but VS Code always suggests paths like `fp-ts/lib/Module`. Does anyone know how to configure it to use unprefixed module paths by default?

![image](https://user-images.githubusercontent.com/7478957/169034465-08bc0b27-ed18-430d-9178-6e3e599f3cef.png)

One extra question: is there any way to configure automatic _namespaced_ imports? So if I wrote `map`, it would import it like `import * as Option from fp-ts/Option`. I'm really annoyed when I have to type imports manually.

**Comment by samhh** [comment]
You'll need to configure it for each module but at Unsplash we built [`ts-namespace-import-plugin`](https://github.com/unsplash/ts-namespace-import-plugin) for the latter issue. If you write for example `Option` then it can suggest an auto-import for the namespace.

**Reply by onvlt** [reply]
This is nice. It would be nice if it allowed some kind of wildcards so one wouldn't need to configure each module separately. But it will do the job.

There are also few VS Code extensions for namespaced auto-imports ([ts-namespace-imports](https://marketplace.visualstudio.com/items?itemName=MikeCann.ts-namespace-imports), [typescript-namespace-imports-vscode-plugin](https://marketplace.visualstudio.com/items?itemName=josh211ua.typescript-namespace-imports-vscode-plugin)), but they don't seem to work for me for some reason.

**Comment by steida** [comment]
```import { either, readonlyArray } from "fp-ts";```
  
You can auto-import modules from 'fp-ts'. As for types, I manually remove `/lib` 🤷‍♂️

**Reply by onvlt** [reply]
Oh, I didn't know I could import whole namespaced modules directly from `fp-ts` like this. But the problem is again that when I type `either`, language service doesn't suggest `import { either } from "fp-ts"`, but `import { either } from "fp-ts/lib/Either"` (which is typeclass instance, not module), so I have to type import manually anyway.

When importing types, manually removing `lib` is exactly what I do and what I want to avoid. It works fine, but sometimes I forget to remove it. One could write eslint rule for this, but this wouldn't be necessary if TypeScript imported correct versions by itself.

EDIT: Ok, I figured out VS Code starts suggesting imports from `fp-ts` when I have already some imports like this in my project. So this is partial solution.

**Reply by steida** [reply]
@vanniewelt As you wrote, auto import from fp-ts without `/lib` is suggested. Except for types, but it should be OK, because types are removed anyway, so it's only an aesthetical issue.
